### PR TITLE
Release code + scripts

### PR DESCRIPTION
**Pull request by jonahjon**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: generatebundlefile/promote.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		clients.ecrPublicClientRelease, err = NewECRPublicClient(true, &cfg)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to create SDK connection to ECR Public another profile %s", err)
+		}
+	}
+	return clients, nil
+}
+
+// PromoteHelmChart will take a given repository, and authFile and handle helm and image promotion for the mentioned chart.
+func (c *SDKClients) PromoteHelmChart(repository, authFile string, profile bool) error {
```
Comment (ewollesen) [inline generatebundlefile/promote.go:65]: Should `profile` be named `public`?

File: generatebundlefile/promote.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	// Change the source destination check depending on release or not
+	destination := c.ecrPublicClient
+	if profile {
+		destination = c.ecrPublicClientRelease
+	}
+
+	// Loop through each image, and the helm chart itself and check for existance in ECR Public, skip if we find the SHA already exists in destination.
+	// If we don't find the SHA in public, we lookup the tag from Private, and copy from private to Public with the same tag.
+	for _, images := range helmRequires.Spec.Images {
+		check, err := destination.shaExistsInRepository(images.Repository, images.Digest)
```
Comment (ewollesen) [inline generatebundlefile/promote.go:129]: Is it really ok if all of these fail? You log some things, but you don't error out. The user could potentially miss the logs and assume that everything was ok.

File: generatebundlefile/ecr.go [hunk 3]
```diff
@@ -40,12 +40,17 @@ type ecrClient struct {
...
 	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(ecrPublicRegion))
 	if err != nil {
 		return nil, fmt.Errorf("Creating AWS ECR Public config %w", err)
 	}
-	ecrPublicClient := &ecrPublicClient{Client: ecrpublic.NewFromConfig(cfg)}
+	ecrPublicClient := &ecrPublicClient{}
+	if conf != nil {
+		ecrPublicClient.Client = ecrpublic.NewFromConfig(*conf)
+	} else {
+		ecrPublicClient.Client = ecrpublic.NewFromConfig(cfg)
```
[the remark this answers is not in the document]
Comment (jonahjon) [reply generatebundlefile/ecr.go:52]: Looks good, got it tested, and works good.

File: generatebundlefile/ecr.go [hunk 4]
```diff
@@ -410,25 +454,50 @@ func splitECRName(s string) (string, string, error) {
...
 		ecrname := fmt.Sprintf("%s.dkr.ecr.us-west-2.amazonaws.com/%s", accountID, name)
-		return ecrname, version, err
+		return ecrname, version, sha, err
+	}
+	version = splitname[1]
+	return name, version, sha, nil
+}
+
+// getNameAndVersionPublic looks up the latest pushed helm chart's tag from a given repo name in Public ECR OCI format.
+func (c *SDKClients) getNameAndVersionPublic(s, registryURI string) (string, string, string, error) {
```
[the remark this answers is not in the document]
Comment (jonahjon) [reply generatebundlefile/ecr.go:482]: I fixed the comment, and variable name to make more sense so a user doesn't supply the wrong input.

File: generatebundlefile/promote.go (hunk 1 above)
Comment (jonahjon) [reply generatebundlefile/promote.go:65]: It's about whether or not it's using a profile to perform cross account promotion, I will change to be more clear.

File: generatebundlefile/promote.go (hunk 2 above)
Comment (jonahjon) [reply generatebundlefile/promote.go:127]: Added return errors for these in commit.

**Comment by TerryHowe** [comment]
Is this g2g?

**Comment by jonahjon** [comment]
> Is this g2g?

I think so, Waiting on Eric but we could just send it too.

**Review by ewollesen — COMMENTED** [review]
/lgtm
/approve

**Review by ewollesen — APPROVED** [review]

**Pull request merged**